ext/src/Map.c

```diff
@@ -518,11 +518,14 @@ php_driver_map_compare(zval *obj1, zval *obj2 TSRMLS_DC)
   }
 
   HASH_ITER(hh, map1->entries, curr, temp) {
-    php_driver_map_entry *entry;
+    php_driver_map_entry *entry = NULL;
     HASH_FIND_ZVAL(map2->entries, PHP5TO7_ZVAL_MAYBE_P(curr->key), entry);
     if (entry == NULL) {
       return 1;
     }
```

Comment:
For better or worse. I just discovered that PHP's internal hash table does the same thing for unordered arrays with keys: https://github.com/php/php-src/blob/master/Zend/zend_hash.c#L2352-L2364